### PR TITLE
HR attendance refactor

### PR DESCRIPTION
**Pull request by ecino**
Since the name of the module is changed, I added a symlink for the migration, we can remove it afterwards.

### CODE REVIEW

File: hr_attendance_calendar/models/hr_attendance.py [hunk 1]
```diff
@@ -37,43 +35,51 @@ def create(self, vals):
...
+            ])
+            if not attendance_day:
+                attendance_day = self.env['hr.attendance.day'].create({
+                    'employee_id': employee_id,
+                    'date': date
+                })
+            else:
+                # A modified attendance should update related breaks
+                attendance_day.compute_breaks()
+            attendance_days |= attendance_day
```
Comment (vwvw) [inline hr_attendance_calendar/models/hr_attendance.py:84]: attendance_days.update(attendance_day) seems clearer to me.

File: hr_attendance_calendar/models/hr_attendance_day.py [hunk 2]
```diff
@@ -154,21 +151,11 @@ def _compute_due_hours(self):
...
-                for leave in att_day.leave_ids.filtered(
-                        lambda l: l.state in ('validate', 'validate1')):
-                    if not leave.holiday_status_id.remove_from_due_hours:
-                        working_day = self.env[
-                            'base.config.settings'].get_work_day_duration()
-                        due_hours -= leave.compute_work_day(
-                            att_day.date) * working_day
+            for leave in att_day.leave_ids.filtered(
+                    lambda l: l.state == 'validate' and not l.keep_due_hours):
+                due_hours -= leave.compute_leave_time(att_day.date)
```
Comment (vwvw) [inline hr_attendance_calendar/models/hr_attendance_day.py:158]: compute_leave_time is in range [0-24]. Are you sure that you want to deduce 24 hours of work here? 
In particular, if I leave at noon, leave time will return 12 but due hours should probably be reduced only by half. no?

Comment (ecino) [reply hr_attendance_calendar/models/hr_attendance_day.py:158]: We discussed with David and choose to let it like that. This scenario will almost never happen and if it happens, the only consequence is the employee will have a bit more extra hours. The manager can correct this manually if needed (or can catch it before approving the leave request).

File: hr_attendance_calendar/models/hr_attendance.py (hunk 1 above)
Comment (ecino) [reply hr_attendance_calendar/models/hr_attendance.py:84]: Not producing the same result. Update method on recordset is something else...

**Comment by ecino** [comment]
I will add the tests directly in this PR

File: hr_attendance_calendar/models/hr_attendance_day.py (hunk 2 above)
Comment (vwvw) [reply hr_attendance_calendar/models/hr_attendance_day.py:158]: Yup sounds good to me then !

**Review by eicher31 — APPROVED** [review]

**Pull request merged**